### PR TITLE
Adding more parameters to kmeans and basic unit tests

### PR DESCRIPTION
**Pull request by damirpolat**
I added two more parameters for kmeans -- `iter.max` and `algorithm` and allowed `centers` to be either `integer` or a `data.frame` following `stats::kmeans` documentation. 

I also added basic unit tests adopted from mlr3. 

Thanks,
Damir

### CODE REVIEW

**Comment by mllg** [comment]
It is not necessary to copy over the tests/expectations from mlr3. Here is an example showing how to load them directly from the mlr3 install directory:

https://github.com/mlr-org/mlr3filters/blob/master/tests/testthat/helper.R#L3

File: tests/testthat/helper.R [hunk 1]
```diff
@@ -0,0 +1,6 @@
+library(mlr3)
+library(mlr3cluster)
+library(checkmate)
+library(testthat)
+
+lapply(list.files(system.file("testthat", package = "mlr3cluster"), pattern = "^helper.*\\.[rR]", full.names = TRUE), source)
```
Comment (mllg) [inline tests/testthat/helper.R:6]: Note that this sources files from "[mlr3cluster]/inst/testthat". You can remove this line.

File: tests/testthat/setup.R [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  warnPartialMatchArgs = TRUE,
+  warnPartialMatchAttr = TRUE,
+  warnPartialMatchDollar = TRUE
+)
+
+# https://github.com/HenrikBengtsson/Wishlist-for-R/issues/88
+old_opts = lapply(old_opts, function(x) if (is.null(x)) FALSE else x)
+
+old_plan = future::plan()
+future::plan("sequential")
```
Comment (mllg) [inline tests/testthat/setup.R:11]: Is the future package necessary for mlr3cluster? I think you can remove these lines and also update "Suggests" in the DESCRIPTION.

File: tests/testthat/helper.R (hunk 1 above)
Comment (damirpolat) [reply tests/testthat/helper.R:6]: Ok, that makes sense.

File: tests/testthat/setup.R (hunk 2 above)
Comment (damirpolat) [reply tests/testthat/setup.R]: I copy/pasted this from mlr3 tests without thinking. I removed this now but I don't think future is necessary for mlr3clutser.

**Pull request merged**